shell/plugins/steve/worker/web-worker.advanced.js

```diff
@@ -3,17 +3,88 @@
  * relocates cluster resource sockets off the UI thread and into a webworker
  */
 
-import { SCHEMA, COUNT } from '@shell/config/types';
+import {
+  SCHEMA, COUNT, POD, WORKLOAD, WORKLOAD_TYPES
+} from '@shell/config/types';
 import ResourceWatcher, { watchKeyFromMessage } from '@shell/plugins/steve/resourceWatcher';
 import { EVENT_MESSAGE, EVENT_CONNECT_ERROR, EVENT_DISCONNECT_ERROR } from '@shell/utils/socket';
 import { normalizeType, keyFieldFor } from '@shell/plugins/dashboard-store/normalize';
-import { addSchemaIndexFields } from '@shell/plugins/steve/schema.utils';
+import { addSchemaIndexFields } from '@shell/plugins/steve/resourceUtils/schema';
 import cacheClasses from '@shell/plugins/steve/caches';
 import ResourceRequest from '@shell/plugins/steve/api/resourceRequest';
 import Trace from '@shell/plugins/steve/trace';
+import { CACHE_STATES } from '@shell/plugins/steve/caches/base-cache';
+import { hashObj } from '~/shell/utils/crypto/browserHashUtils';
 
 const caches = {};
 
+// TODO: RC TEST - workloads. each pods
+// TODO: RC TEST - configmap search --> crt
+// TODO: RC TEST - run through all cluster models and check for super?
+// TODO: RC TEST - everything turned off
+// TODO: RC test podsByNamespace
+// TODO: RC test Work with intermedate loading / manual / project&ns forced filtering
+// TODO: RC test resourceUtils vs place they come from. Includes
+// - getters, etc.
+// - caches in calculated fields correct
+// TODO: RC test resourceUti
+
+const uiRequests = {};
+const uiApi = (getter) => {
+  const cacheGetter = `${ getter }/cacheRequest`;
+  const cacheAge = Date.now() - caches[getter].loadedTimestamp || 0;
+  const uiPromise = {};
+
+  uiPromise.promise = new Promise((resolve, reject) => {
+    uiPromise.resolves = ({ cache }) => {
+      resolve({ data: { data: cache } });
+    };
+    uiPromise.reject = (error) => {
+      reject(error);
+    };
+
+    if (cacheAge > 500 && (uiRequests[cacheGetter] || []).length < 1) {
+      // TODO: RC question If the cache of non-cluster resources changes... do we ned to bump anything that depends on them?
+      self.postMessage({ get: cacheGetter });
```

Comment:
Do we ever rely on content from outside the cluster store in any of the calculated fields?